### DISCUSSION TITLE
Count of total and failed tests is off

### DISCUSSION THREAD

**Discussion by josdejong** [vitest-dev/vscode#417] [Q&A]
I just started using the this Vitest extension in VS Code, and noticed that the total count of tests and successful tests is off. For example in the following screenshot, there is 1 test failing out of 430 tests, but the totals of the plugin suggests 4 failing tests out of 533 which is quite misleading.

Here is a minimal example demonstrating the issue:

```js
import { test, describe, expect } from 'vitest'

function add(a, b) {
  return a + b
}

describe('describe A', () => {
  describe('describe B', () => {
    test('test 1', () => {
      expect(add(2, 2)).toBe(4)
    })

    test('test 2', () => {
      expect(add(3, 3)).toBe(42) // This test will fail
    })
  })
})
```

This example contains two tests inside 2 nested `describe` statements, where 1 of the tests fails. Running this with the test plugin I get:

![image](https://github.com/vitest-dev/vscode/assets/568626/20130123-b468-476a-9285-761126161351)

Which suggests there are 4 failing tests out of 5 (but it is 1 failing test out of 2)

On the command line, the counts are correct:

```
 RERUN  rerun all tests x5

 ❯ index.test.js (2)
   ❯ describe A (2)
     ❯ describe B (2)
       ✓ test 1
       × test 2

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 1 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

...

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯
 Test Files  1 failed (1)
      Tests  1 failed | 1 passed (2)
   Start at  09:40:25
   Duration  10ms


 FAIL  Tests failed. Watching for file changes...
       press h to show help, press q to quit
```

It looks like the `describe(...)` wrappers are also counted as a test too, and I don't think that should be the case. 

I guess this is a bug?

**Comment by sheremet-va** [comment]
This is how vscode's built-in Test API works. It shows the tree up to the root if there is a failing test so it's easier to find if you have tests collapsed. I think it is quite useful and intended.

**Reply by sheremet-va** [reply]
Note: this is also how it works in Vitest. If the suite has at least one failing test, the suite is considered failed.

**Reply by josdejong** [reply]
Thanks for your input. The red icons on the parent folders of a failed test are indeed helpful and needed to find a nested, failed test, but the numbers are wrong. I understand that the suite is marked as failed when one test inside it fails, but it is confusing and not helpful at all to add the number of failed suites to the number of failed tests and present them as the number of failed tests.

The Vitest VS Code plugin  says "1/5 tests passed" on top and in the icon on the left "4 failed tests", that doesn't make sense to me. The reality is "1/2 tests passed" and "1 failed test". The Vitest command line outputs the right numbers: "Tests  1 failed | 1 passed (2)".

So is the issue that it may be hard to fix this? Or so you see this as intended and correct behavior?

**Reply by sheremet-va** [reply]
This does seem like a VSCode bug because both values are reported by vscode itself, not Vitest.

**Reply by josdejong** [reply]
I converted the example project to Jest and used the Jest test runner, that shows correct numbers, "1/2 tests passed", and "1 failing test":

![image](https://github.com/vitest-dev/vscode/assets/568626/322891b8-ccf8-46e6-9cb0-a3e928e381c1)

So it looks like this issue is specific to the vitest plugin and not an issue of the VS Code Test Runner in general.

**Reply by sheremet-va** [reply]
I can look into it if you can provide a simple reproduction in the form of a repository/stackblitz link that I can install on my machine and run.

**Reply by josdejong** [reply]
Thanks! Here you go: https://github.com/josdejong/vitest-vscode-count-issue

**Reply by sheremet-va** [reply]
Thank you! I fixed the error in #419

**Reply by josdejong** [reply]
Awesome! Thanks Vladimir.

**Reply by josdejong** [reply]
I can confirm that the latest version of the plugin now shows the right numbers 🎉

**Reply by josdejong** [reply]
There is one thing that still could be improved: whilst the tests are running, the old total (including test suites) is shown. When the tests are finished, the right total is shown.
